static/js/services/add-attachment.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Service to create and attach data to a couchdb doc
+ */
+angular.module('inboxServices').factory('AddAttachment',
+  function() {
+    'ngInject';
+    'use strict';
+
+    return function(doc, name, content, contentType) {
+      if (!doc._attachments) {
+        doc._attachments = {};
+      }
+      var blob = new Blob([ content ], { type: contentType });
```

Comment:
You know what? I have no idea! Everyone says to do it and nobody says why. I might ask on SO. My best guess is it's something to do with reading it later. You can turn a blob into a url by calling URL.createObjectUrl so maybe it helps when the browser resolves the URL? I'm sure the default would work just as well in this case...